File: C++/Medium/Accepted/number-of-substrings-with-only-1s_9.cpp
class Solution {
public:
    int numSub(string s) {
        
        long long int count = 0;
        long long int mod = 1000000007;
        long long int sum = 0;
        for(int i = 0; i < s.size(); i++) {
            if(s[i] == '1') {
                count++;
            } else {
                sum = (sum + ((count * (count+1))/2)) % mod;
                count = 0;
            }
        }
        sum = (sum + ((count * (count+1))/2)) % mod;
        return sum;
    }
};

File: Java/Medium/WrongAnswer/number-of-substrings-with-only-1s_9.java
class Solution {
    public int numSub(String s) {
        
        int mod = (int) (1e9 + 7);
        int count = 0;
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '1') {
                count++;
            } else {
                res += (count * (count + 1)) / 2;
                res %= mod;
                count = 0;
            }
        }
        res += (count * (count + 1)) / 2;
        res %= mod;
        return res;
    }
}